dipy/align/_public.py

```diff
@@ -0,0 +1,833 @@
+
+    Parameters
+    ----------
+    moving : array, nifti image or str
+        Containing the data for the moving object, or full path to a nifti file
+        with the moving data.
+
+    moving_affine : 4x4 array, optional
+        An affine transformation associated with the moving object. Required if
+        data is provided as an array. If provided together with nifti/path,
+        will over-ride the affine that is in the nifti.
+
+    static : array, nifti image or str
+        Containing the data for the static object, or full path to a nifti file
+        with the moving data.
+
+    static_affine : 4x4 array, optional
+        An affine transformation associated with the static object. Required if
+        data is provided as an array. If provided together with nifti/path,
+        will over-ride the affine that is in the nifti.
+
+    between_affine: 4x4 array, optional
+        If an additional affine is needed betweeen the two spaces.
+        Default: identity (no additional registration).
+
+    Returns
+    -------
+    A Nifti1Image class instance with the data from the moving object
+    resampled into the space of the static object.
+    """
+
+    static, static_affine, moving, moving_affine, between_affine = \
+        _handle_pipeline_inputs(moving, static,
+                                moving_affine=moving_affine,
+                                static_affine=static_affine,
+                                starting_affine=between_affine)
+    affine_map = AffineMap(between_affine,
+                           static.shape, static_affine,
+                           moving.shape, moving_affine)
+    resampled = affine_map.transform(moving)
+    return nib.Nifti1Image(resampled, static_affine)
+
+
+def c_of_mass(moving, static, static_affine=None, moving_affine=None,
+              starting_affine=None, reg=None):
+    """
+    Implements a center of mass transform
+
+    Parameters
+    ----------
+    moving : array, nifti image or str
+        Containing the data for the moving object, or full path to a nifti file
+        with the moving data.
+
+    moving_affine : 4x4 array, optional
+        An affine transformation associated with the moving object. Required if
+        data is provided as an array. If provided together with nifti/path,
+        will over-ride the affine that is in the nifti.
+
+    static : array, nifti image or str
+        Containing the data for the static object, or full path to a nifti file
+        with the moving data.
+
+    static_affine : 4x4 array, optional
+        An affine transformation associated with the static object. Required if
+        data is provided as an array. If provided together with nifti/path,
+        will over-ride the affine that is in the nifti.
+
+    starting_affine: 4x4 array, optional
+        Initial guess for the transformation between the spaces.
+
+    reg : not needed here. Use None
```

Comment:
So why do we keep this unused parameter?

Reply:
So that we can use the same call to different pipeline components.